edb/server/compiler/sertypes.py

```diff
@@ -38,6 +37,8 @@
 from edb.schema import scalars as s_scalars
 from edb.schema import types as s_types
 
+from edb.testbase import protocol
```

Comment:
Importing this from `testbase` here is a bit weird.  I think we should move the definition here and import it into `testbase/protocol/messages` instead.

Reply:
changed

Reply:
Let's also get rid of `ResultCardinality` and change `render_cardinality` to use the consolidated enum

Reply:
changed

Reply:
Hm, I don't see changes to `render_cardinality` in `binary.pyx` in the diff still.